litho-core/src/main/java/com/facebook/litho/ComponentTree.java

```diff
@@ -2143,7 +2135,18 @@ public LayoutState get() {
         }
       }
 
-      return layoutState.acquireRef();
+      final int calculatedLayoutStateId = layoutState.getId();
+      if (layoutStateId.compareAndSet(STATE_UNINITIALIZED, calculatedLayoutStateId)) {
+        // This is the first get(), we don't need to acquireRef because refcount is already 1.
+        return layoutState;
+      } else if (calculatedLayoutStateId == layoutStateId.get()) {
+        // Subsequent get() calls need to increment refcount.
+        return layoutState.acquireRef();
```

Comment:
should probably wrap this in a try/catch still, in case it gets released right before this